src/main/java/duke/Event.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Represents an event that will happen. Has a description, date
+ * that the event occurs, and can be marked as done or undone.
+ */
+public class Event extends Task {
+    protected LocalDate on;
+
+    public Event(String description, String on) {
+        this(description, on, false);
+    }
+
+    public Event(String description, String on, boolean isDone) {
+        super(description, 'E', isDone);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        this.on = LocalDate.parse(Arrays.stream(on.split(" ")).skip(1)
+                .collect(Collectors.joining("")), formatter);
+    }
+
+    /**
+     * Converts save string data to a Event object.
+     * The save string data is in the format:
+     * <p>
+     * <pre>
+     * E,"<description>","<by>"
+     * </pre>
+     * <p>
+     * 
+     * @param saveString the save string data
+     * @return the new Event object created from saveString
+     * @throws DukeException
+     */
+    public static Event fromSaveString(String saveString) throws DukeException {
+        String[] splitSaveString = saveString.split("(\",\")|(\",)|(,\")|\"");
+        if(splitSaveString.length != 3) {

```

Comment:
It would be good if you had a spacing between the if and open bracket here as well.